app/src/common/shared/com/igalia/wolvic/input/MotionEventGenerator.java

```diff
@@ -86,7 +87,7 @@ private static void generateEvent(Widget aWidget, Device aDevice, boolean aFocus
         event.recycle();
     }
 
-    public static void dispatch(Widget aWidget, int aDevice, boolean aFocused, boolean aPressed, float aX, float aY) {
+    public static void dispatch(WidgetManagerDelegate widgetManager, Widget aWidget, int aDevice, boolean aFocused, boolean aPressed, float aX, float aY) {
```

Comment:
I don't think we should change this. If we want to trigger haptic feedback in the cases you change bellow then we better listen for the events we generate in those cases and act accordingly in the event handlers.

Reply:
Actually, I prefer this way personally. If we choose to handle this at the widgets side, we will need to make sure that all the widgets have access to widgetManager and all the widgets listen to the hover events, which will need us to duplicate a lot of code.